### PR TITLE
fixing APIMANAGER-3953 

### PR DESCRIPTION
**Pull request by ruks**
fixed issue 3953

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.usage/org.wso2.carbon.apimgt.usage.client/src/main/java/org/wso2/carbon/apimgt/usage/client/APIUsageStatisticsClient.java [hunk 1]
```diff
@@ -447,11 +447,24 @@ public static OMElement buildOMElement(InputStream inputStream) throws Exception
             String query = "SELECT CONSUMER_KEY, NAME FROM AM_APPLICATION_KEY_MAPPING INNER JOIN AM_APPLICATION ON " +
                            "AM_APPLICATION_KEY_MAPPING.APPLICATION_ID=AM_APPLICATION.APPLICATION_ID INNER JOIN " +
                            "AM_SUBSCRIBER" +
-                           " ON AM_APPLICATION.SUBSCRIBER_ID = AM_SUBSCRIBER.SUBSCRIBER_ID WHERE AM_SUBSCRIBER" +
-                           ".USER_ID = ? ";
+                           " ON AM_APPLICATION.SUBSCRIBER_ID = AM_SUBSCRIBER.SUBSCRIBER_ID WHERE ";
+
+            boolean sharedApp;
+            if (groupId != null && !"".equals(groupId)) {
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.usage/org.wso2.carbon.apimgt.usage.client/src/main/java/org/wso2/carbon/apimgt/usage/client/APIUsageStatisticsClient.java]: you may ise StringUtils.isEmpty

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/site/themes/fancy/templates/subscription/subscription-list/js/subscription-list.js [hunk 2]
```diff
@@ -445,20 +445,17 @@ var regenerate=function(appName,keyType,i,btn,div,clientId,clientSecret) {
...
+                                        '</div>'+
+                                        '<div class="span9">'+
+                                        '<div class="token">'+
+                                        '<span class="accessTokenScopeDisplayPro keyValues" id="prodAccessScope" data-value="'+generatedScopesNames+'">'+generatedScopesNames+'</span>'+
+                                        '</div>'+
+                                        '</div>'+
+                                        '</div>';
+                   tokenTxtBox.before(scopeTxtDiv);
                 }
             }else{  // if Scope is not selected
```
Comment (ChamNDeSilva) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/site/themes/fancy/templates/subscription/subscription-list/js/subscription-list.js]: Please fix the formatting issue : line 460 and 461

File: components/apimgt/org.wso2.carbon.apimgt.usage/org.wso2.carbon.apimgt.usage.client/src/main/java/org/wso2/carbon/apimgt/usage/client/APIUsageStatisticsClient.java [hunk 3]
```diff
@@ -152,14 +152,14 @@ public static OMElement buildOMElement(InputStream inputStream) throws Exception
         return builder.getDocumentElement();
     }
 
-    public List<APIResponseFaultCountDTO> getPerAppFaultCount(String subscriberName, String fromDate, String toDate, int limit)
+    public List<APIResponseFaultCountDTO> getPerAppFaultCount(String subscriberName, String groupId, String fromDate, String toDate, int limit)
```
Comment (chamindias) [inline components/apimgt/org.wso2.carbon.apimgt.usage/org.wso2.carbon.apimgt.usage.client/src/main/java/org/wso2/carbon/apimgt/usage/client/APIUsageStatisticsClient.java]: Shall we add method comments to all the missing places in this PR? (Line 155,193,249,311,399,439)

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/site/themes/fancy/templates/subscription/subscription-list/js/subscription-list.js [hunk 4]
```diff
@@ -445,20 +445,17 @@ var regenerate=function(appName,keyType,i,btn,div,clientId,clientSecret) {
...
-                                    '</div>'+
-                                    '<div class="span9">'+
-                                    '<div class="token">'+
-                                    '<span class="accessTokenScopeDisplayPro keyValues" id="prodAccessScope" data-value="'+generatedScopesNames+'">'+generatedScopesNames+'</span>'+
-                                    '</div>'+
-                                    '</div>'+
-                                    '</div>';
-                        }
-                    );
+                    var scopeTxtDiv=    '<div class="row-fluid keys">'+
```
Comment (ChamNDeSilva) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/site/themes/fancy/templates/subscription/subscription-list/js/subscription-list.js]: formatting issue.

File: components/apimgt/org.wso2.carbon.apimgt.usage/org.wso2.carbon.apimgt.usage.client/src/main/java/org/wso2/carbon/apimgt/usage/client/APIUsageStatisticsClient.java (hunk 1 above)
Comment (ruks) [inline components/apimgt/org.wso2.carbon.apimgt.usage/org.wso2.carbon.apimgt.usage.client/src/main/java/org/wso2/carbon/apimgt/usage/client/APIUsageStatisticsClient.java]: fixed in commit: https://github.com/wso2/carbon-apimgt/commit/9619da2271b49f37ef93456b6f8a8ac235f685ea

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/site/themes/fancy/templates/subscription/subscription-list/js/subscription-list.js (hunk 4 above)
Comment (ruks) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/site/themes/fancy/templates/subscription/subscription-list/js/subscription-list.js]: fixed in commit: https://github.com/wso2/carbon-apimgt/commit/d7e56b1d85254369d07aab9572f13d4c2120df97

File: components/apimgt/org.wso2.carbon.apimgt.usage/org.wso2.carbon.apimgt.usage.client/src/main/java/org/wso2/carbon/apimgt/usage/client/APIUsageStatisticsClient.java (hunk 3 above)
Comment (ruks) [inline components/apimgt/org.wso2.carbon.apimgt.usage/org.wso2.carbon.apimgt.usage.client/src/main/java/org/wso2/carbon/apimgt/usage/client/APIUsageStatisticsClient.java]: fixed in PR: https://github.com/wso2/carbon-apimgt/pull/1582

File: components/apimgt/org.wso2.carbon.apimgt.usage/org.wso2.carbon.apimgt.usage.client/src/main/java/org/wso2/carbon/apimgt/usage/client/APIUsageStatisticsClient.java [hunk 5]
```diff
@@ -308,10 +308,10 @@ public static OMElement buildOMElement(InputStream inputStream) throws Exception
 
     }
 
-    public List<AppRegisteredUsersDTO> getAppRegisteredUsers(String subscriberName) throws APIMgtUsageQueryServiceClientException {
+    public List<AppRegisteredUsersDTO> getAppRegisteredUsers(String subscriberName, String groupId) throws APIMgtUsageQueryServiceClientException {
```
Comment (sinthuja) [inline components/apimgt/org.wso2.carbon.apimgt.usage/org.wso2.carbon.apimgt.usage.client/src/main/java/org/wso2/carbon/apimgt/usage/client/APIUsageStatisticsClient.java]: Remove the unnecessary blank lines.

Comment (ruks) [inline components/apimgt/org.wso2.carbon.apimgt.usage/org.wso2.carbon.apimgt.usage.client/src/main/java/org/wso2/carbon/apimgt/usage/client/APIUsageStatisticsClient.java]: @sinthuja this changes are moved to
 org.wso2.carbon.apimgt.usage.client.impl.APIUsageStatisticsRdbmsClientImpl.java
org.wso2.carbon.apimgt.usage.client.impl.APIUsageStatisticsRestClientImpl.java classes. hence fixed in there in commit: https://github.com/wso2/carbon-apimgt/commit/ef30cea43f77c9b6f6500f3f70449d7238a942e4

**Pull request merged**